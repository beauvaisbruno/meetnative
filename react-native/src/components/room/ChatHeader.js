"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const native_base_1 = require("native-base");
const Avatars_1 = require("../profile/img/Avatars");
const react_native_responsive_screen_1 = require("react-native-responsive-screen");
const react_native_auto_height_image_1 = __importDefault(require("react-native-auto-height-image"));
const utils_1 = require("../../utils/utils");
const utilsUi_1 = require("../../utils/utilsUi");
const lang_1 = __importDefault(require("./lang"));
const custom_1 = __importDefault(require("../../config/native-base-theme/variables/custom"));
class ChatHeader extends React.Component {
    constructor(props) {
        super(props);
        this.state = Object.assign({ headerOpen: false }, props.storyState);
    }
    render() {
        const s = utilsUi_1.font(10);
        const m = utilsUi_1.font(15);
        const l = utilsUi_1.font(20);
        const xl = utilsUi_1.font(27);
        const profile = this.props.profile;
        const user = this.props.user;
        return (React.createElement(native_base_1.View, null,
            !this.state.headerOpen && (React.createElement(native_base_1.View, { style: {
                    flexDirection: "row",
                    justifyContent: "space-between",
                    borderBottomWidth: 1,
                    borderBottomColor: custom_1.default.borderColor
                } },
                React.createElement(native_base_1.View, { style: {
                        flexDirection: "row"
                    } },
                    React.createElement(react_native_auto_height_image_1.default, { source: Avatars_1.avatars[profile.avatar], width: react_native_responsive_screen_1.widthPercentageToDP("13%") }),
                    React.createElement(native_base_1.Text, { style: { marginLeft: 5, fontWeight: "bold", fontSize: xl } }, profile.pseudo)),
                React.createElement(native_base_1.Button, { block: true, transparent: true, onPress: () => this.handleSetHeaderVisible(true), testID: "headerVisibleBtn" },
                    React.createElement(native_base_1.Icon, { name: "plus-circle", type: "FontAwesome", fontSize: 10, style: { marginLeft: 0, marginRight: 0, fontSize: 10 } }),
                    React.createElement(native_base_1.Text, { style: { fontSize: s, paddingLeft: 5, paddingRight: 5 } }, lang_1.default("Info"))))),
            this.state.headerOpen && (React.createElement(native_base_1.View, { style: {
                    flexDirection: "row",
                    borderBottomWidth: 1,
                    borderBottomColor: custom_1.default.borderColor
                } },
                React.createElement(react_native_auto_height_image_1.default, { source: Avatars_1.avatars[profile.avatar], width: react_native_responsive_screen_1.widthPercentageToDP("50%") }),
                React.createElement(native_base_1.View, { style: { paddingLeft: 5, width: react_native_responsive_screen_1.widthPercentageToDP("50%") } },
                    React.createElement(native_base_1.View, { style: {
                            flexDirection: "row",
                            justifyContent: "space-between",
                            alignItems: "center"
                        } },
                        React.createElement(native_base_1.View, { style: { flexDirection: "row" } },
                            React.createElement(native_base_1.Text, { style: { fontWeight: "bold", fontSize: l } }, profile.pseudo),
                            React.createElement(native_base_1.Icon, { name: profile.isMale ? "gender-male" : "gender-female", type: "MaterialCommunityIcons", style: {
                                    color: profile.isMale
                                        ? custom_1.default.brandPrimary
                                        : custom_1.default.brandSecondary
                                }, fontSize: l })),
                        React.createElement(native_base_1.Button, { block: true, transparent: true, onPress: () => this.handleSetHeaderVisible(false), testID: "headerVisibleBtn", style: Object.assign({}, utilsUi_1.noMarPad, { height: m + 5 }) },
                            React.createElement(native_base_1.Icon, { name: "close-circle-outline", type: "MaterialCommunityIcons", style: Object.assign({}, utilsUi_1.noMarPad, { fontSize: m, color: custom_1.default.brandDanger }) }))),
                    React.createElement(native_base_1.Text, { style: { color: custom_1.default.brandPrimary, fontSize: m } },
                        profile.city,
                        " ",
                        utils_1.distanceKm(user.lat, user.lon, profile.lat, profile.lon),
                        "km"),
                    React.createElement(native_base_1.Text, { style: { color: custom_1.default.textGreyLight, fontSize: s } },
                        profile.profession,
                        " - ",
                        profile.age + lang_1.default("yo")),
                    React.createElement(native_base_1.Text, { style: { fontWeight: "bold", fontSize: s } }, lang_1.default("Speak")),
                    utilsUi_1.teachRows(user, s, profile.langsToLearn),
                    React.createElement(native_base_1.Text, { style: { fontWeight: "bold", fontSize: s } }, lang_1.default("Learn")),
                    utilsUi_1.learnRows(user, s, profile.langsToTeach))))));
    }
    handleSetHeaderVisible(headerOpen) {
        this.setState({ headerOpen });
    }
}
exports.ChatHeader = ChatHeader;
exports.default = ChatHeader;
