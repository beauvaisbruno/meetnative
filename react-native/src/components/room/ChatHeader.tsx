import { IProfile } from "../../rootReducers";
import * as React from "react";
import { Button, Content, Icon, Text, View } from "native-base";
import { avatars } from "../profile/img/Avatars";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import AutoHeightImage from "react-native-auto-height-image";
import { distanceKm } from "../../utils/utils";
import { font, learnRows, noMarPad, teachRows } from "../../utils/utilsUi";
import lang from "./lang";
import custom from "../../config/native-base-theme/variables/custom";

export type IProps = {
  profile: IProfile;
  user: IProfile;
  storyState?: object;
};
export interface IState {
  headerOpen: boolean;
}

export class ChatHeader extends React.Component<IProps, IState> {
  constructor(props: any) {
    super(props);
    this.state = {
      headerOpen: false,
      ...props.storyState
    };
  }
  render() {
    const s = font(10);
    const m = font(15);
    const l = font(20);
    const xl = font(27);
    const profile = this.props.profile;
    const user = this.props.user;
    return (
      <View>
        {!this.state.headerOpen && (
          <View
            style={{
              flexDirection: "row",
              justifyContent: "space-between",
              borderBottomWidth: 1,
              borderBottomColor: custom.borderColor
            }}
          >
            <View
              style={{
                flexDirection: "row"
              }}
            >
              <AutoHeightImage
                source={avatars[profile.avatar] as number}
                width={wp("13%")}
              />
              <Text style={{ marginLeft: 5, fontWeight: "bold", fontSize: xl }}>
                {profile.pseudo}
              </Text>
            </View>
            <Button
              block
              transparent
              onPress={() => this.handleSetHeaderVisible(true)}
              testID={"headerVisibleBtn"}
            >
              <Icon
                name="plus-circle"
                type="FontAwesome"
                fontSize={10}
                style={{ marginLeft: 0, marginRight: 0, fontSize: 10 }}
              />
              <Text style={{ fontSize: s, paddingLeft: 5, paddingRight: 5 }}>
                {lang("Info")}
              </Text>
            </Button>
          </View>
        )}
        {this.state.headerOpen && (
          <View
            style={{
              flexDirection: "row",
              borderBottomWidth: 1,
              borderBottomColor: custom.borderColor
            }}
          >
            <AutoHeightImage
              source={avatars[profile.avatar] as number}
              width={wp("50%")}
            />
            <View style={{ paddingLeft: 5, width: wp("50%") }}>
              <View
                style={{
                  flexDirection: "row",
                  justifyContent: "space-between",
                  alignItems: "center"
                }}
              >
                <View style={{ flexDirection: "row" }}>
                  <Text style={{ fontWeight: "bold", fontSize: l }}>
                    {profile.pseudo}
                  </Text>
                  <Icon
                    name={profile.isMale ? "gender-male" : "gender-female"}
                    type="MaterialCommunityIcons"
                    style={{
                      color: profile.isMale
                        ? custom.brandPrimary
                        : custom.brandSecondary
                    }}
                    fontSize={l}
                  />
                </View>
                <Button
                  block
                  transparent
                  onPress={() => this.handleSetHeaderVisible(false)}
                  testID={"headerVisibleBtn"}
                  style={{ ...noMarPad, height: m + 5 }}
                >
                  <Icon
                    name="close-circle-outline"
                    type="MaterialCommunityIcons"
                    style={{
                      ...noMarPad,
                      fontSize: m,
                      color: custom.brandDanger
                    }}
                  />
                </Button>
              </View>
              <Text style={{ color: custom.brandPrimary, fontSize: m }}>
                {profile.city}{" "}
                {distanceKm(user.lat!, user.lon!, profile.lat!, profile.lon!)}km
              </Text>
              <Text style={{ color: custom.textGreyLight, fontSize: s }}>
                {profile.profession}
                {" - "}
                {profile.age + lang("yo")}
              </Text>
              <Text style={{ fontWeight: "bold", fontSize: s }}>
                {lang("Speak")}
              </Text>
              {teachRows(user, s, profile.langsToLearn)}
              <Text style={{ fontWeight: "bold", fontSize: s }}>
                {lang("Learn")}
              </Text>
              {learnRows(user, s, profile.langsToTeach)}
            </View>
          </View>
        )}
      </View>
    );
  }
  private handleSetHeaderVisible(headerOpen: boolean) {
    this.setState({ headerOpen });
  }
}

export default ChatHeader;
