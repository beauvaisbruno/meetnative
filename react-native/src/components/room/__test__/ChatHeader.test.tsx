import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import * as React from "react";
import ChatHeader from "../ChatHeader";


const initialProps = {
  user: {
    langsToLearn: {
      french: "intermediate",
      italian: "beginner",
      spanish: "fluent"
    },
    age: 23,
    lat: 46.144516,
    pseudo: "pseudo",
    lon: 6.081338,
    langsToTeach: {
      english: "native",
      spanish: "fluent"
    },
    geohash: 3661469831164302,
    city: "Genève",
    avatar: "cute",
    profession: "profession"
  },
  profile: {
    langsToLearn: {
      english: "beginner",
      french: "fluent",
      german: "intermediate"
    },
    profession: "Legal Assistant",
    city: "Genève",
    geohash: 872962206,
    langsToTeach: {
      french: "fluent",
      italian: "bilingual",
      spanish: "native"
    },
    lon: 6.107057129141845,
    pseudo: "Maëlle",
    avatar: "pelican",
    lat: 46.08995392215012,
    age: 23,
    id: "IEY3I5fpxQilwKzPt0Sc"
  }
};

it("ChatHeader", () => {
  const tree = shallow(
    // @ts-ignore
    <ChatHeader {...initialProps} />
  );
  expect(toJson(tree)).toMatchSnapshot();
  tree
    .find("Styled(Button)")
    .props()
    // @ts-ignore
    .onPress();
  expect(toJson(tree)).toMatchSnapshot();
});
