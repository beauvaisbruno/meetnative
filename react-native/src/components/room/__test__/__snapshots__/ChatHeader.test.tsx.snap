// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeader 1`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    style={
      Object {
        "borderBottomColor": "#a6a6a6",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Styled(ViewNB)
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <ErrorableImage
        source={
          Object {
            "testUri": "../../../src/components/profile/img/pelican.png",
          }
        }
        width={97.5}
      />
      <Styled(Text)
        style={
          Object {
            "fontSize": 64,
            "fontWeight": "bold",
            "marginLeft": 5,
          }
        }
      >
        Maëlle
      </Styled(Text)>
    </Styled(ViewNB)>
    <Styled(Button)
      block={true}
      onPress={[Function]}
      testID="headerVisibleBtn"
      transparent={true}
    >
      <Styled(Icon)
        fontSize={10}
        name="plus-circle"
        style={
          Object {
            "fontSize": 10,
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
        type="FontAwesome"
      />
      <Styled(Text)
        style={
          Object {
            "fontSize": 24,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Info
      </Styled(Text)>
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`ChatHeader 2`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    style={
      Object {
        "borderBottomColor": "#a6a6a6",
        "borderBottomWidth": 1,
        "flexDirection": "row",
      }
    }
  >
    <ErrorableImage
      source={
        Object {
          "testUri": "../../../src/components/profile/img/pelican.png",
        }
      }
      width={375}
    />
    <Styled(ViewNB)
      style={
        Object {
          "paddingLeft": 5,
          "width": 375,
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Styled(ViewNB)
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "fontSize": 47,
                "fontWeight": "bold",
              }
            }
          >
            Maëlle
          </Styled(Text)>
          <Styled(Icon)
            fontSize={47}
            name="gender-female"
            style={
              Object {
                "color": "#ff9900",
              }
            }
            type="MaterialCommunityIcons"
          />
        </Styled(ViewNB)>
        <Styled(Button)
          block={true}
          onPress={[Function]}
          style={
            Object {
              "height": 40,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            }
          }
          testID="headerVisibleBtn"
          transparent={true}
        >
          <Styled(Icon)
            name="close-circle-outline"
            style={
              Object {
                "color": "#d9534f",
                "fontSize": 35,
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              }
            }
            type="MaterialCommunityIcons"
          />
        </Styled(Button)>
      </Styled(ViewNB)>
      <Styled(Text)
        style={
          Object {
            "color": "#8e7cc3",
            "fontSize": 35,
          }
        }
      >
        Genève
         
        6
        km
      </Styled(Text)>
      <Styled(Text)
        style={
          Object {
            "color": "#919191",
            "fontSize": 24,
          }
        }
      >
        Legal Assistant
         - 
        23yo
      </Styled(Text)>
      <Styled(Text)
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Speak
      </Styled(Text)>
      <Styled(Text)
        key="teach-english"
        style={
          Object {
            "color": "#ff9900",
            "fontSize": 24,
          }
        }
      >
        English Native
      </Styled(Text)>
      <Styled(Text)
        key="teach-spanish"
        style={
          Object {
            "color": "#919191",
            "fontSize": 24,
          }
        }
      >
        Spanish Fluent
      </Styled(Text)>
      <Styled(Text)
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Learn
      </Styled(Text)>
      <Styled(Text)
        key="learn-french"
        style={
          Object {
            "color": "#ff9900",
            "fontSize": 24,
          }
        }
      >
        French Intermediate B1/B2
      </Styled(Text)>
      <Styled(Text)
        key="learn-italian"
        style={
          Object {
            "color": "#ff9900",
            "fontSize": 24,
          }
        }
      >
        Italian Beginner A1/A2
      </Styled(Text)>
      <Styled(Text)
        key="learn-spanish"
        style={
          Object {
            "color": "#ff9900",
            "fontSize": 24,
          }
        }
      >
        Spanish Fluent C1/C2
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
